packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -1607,8 +1607,29 @@ describe('<Autocomplete />', () => {
 
       checkHighlightIs(listbox, 'two');
 
-      // three option is added and autocomplete re-renders, reset the highlight
+      // three option is added and autocomplete re-renders, restore the highlight
       setProps({ options: ['one', 'two', 'three'] });
+      checkHighlightIs(listbox, 'two');
+    });
+
+    it("should reset the highlight when previously highlighted option doesn't exists in new options", () => {
```

Comment:
This test passes on master HEAD, so it's probably not testing the resetting of the option highlight case. Is it passing because autocomplete always resets options in the current Material-UI `Autocomplete` version, regardless of whether the previous highlighted option exists? Can you check?